components/search_results/search_results.test.jsx

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import assert from 'assert';
+
+import {shouldRenderFromPropsAndState} from 'components/search_results/search_results.jsx';
+
+describe('components/SearchResults', () => {
+    describe('shouldRenderFromPropsAndState', () => {
+        const result1 = {test: 'test'};
+        const result2 = {test: 'test'};
+        const results = [result1, result2];
+        const props = {
+            prop1: 'someprop',
+            somearray: [1, 2, 3],
+            results,
+        };
+
+        test('should not render', () => {
+            assert.ok(!shouldRenderFromPropsAndState(props, {...props}, props, {...props}));
+            assert.ok(!shouldRenderFromPropsAndState(props, {...props, prop1: 'someprop'}, props, {...props}));
+            assert.ok(!shouldRenderFromPropsAndState(props, {...props, results: [result1, result2]}, props, {...props}));
+        });
+
+        test('should render', () => {
+            assert.ok(shouldRenderFromPropsAndState(props, {...props, prop1: 'newprop'}, props, {...props}));
+            assert.ok(shouldRenderFromPropsAndState(props, {...props}, props, {...props, prop1: 'newprop'}));
+            assert.ok(shouldRenderFromPropsAndState(props, {...props, results: [result2, result1]}, props, {...props}));
+            assert.ok(shouldRenderFromPropsAndState(props, {...props, results: [result1, result2, {test: 'test'}]}, props, {...props}));
+            assert.ok(shouldRenderFromPropsAndState(props, {...props, somearray: [1, 2, 3]}, props, {...props}));
```

Comment:
Just curious why this case would need a re-render? It seems that `somearray` does not change from the original value. 🤔

Reply:
Good question @marianunez. The reason we expect it to cause a re-render is that `somearray` gets created as a new array and we only do a shallow compare of all props/state with the exception of the `results` prop. Generally shallow compares are good enough because if it's a new object that most of the time means that something has changed (due to the Redux/React pattern) and we save time not doing deeper compares. I added the exception for `results` because it is something that frequently gets re-created as a new array but without any changes (because of the posts reducer being so general).